pipelines/copy-number-variation-vector/gcp/HMM.wdl

```diff
@@ -0,0 +1,335 @@
+            ~{mapq_file} \
+            manifest.txt \
+            ~{gc_content_file} \
+            ~{output_dir} \
+            ~{sample_group_id} \
+            > calculate_mean_coverage_by_GC_~{sample_group_id}.log 2>&1
+    ls -lht
+    tar -zcvf ~{output_dir}.tar.gz ~{output_dir}
+    echo "Numbers: ~{acc_threshold} ~{m_threshold} ~{sample_group_id}"
+  
+  >>>
+  runtime {
+    docker: docker
+    memory: ram
+    disks: "local-disk ${disk} HDD"
+    cpu: cpu
+    preemptible: preemptible
+    zones: runtime_zones
+  }
+  output {
+    #log output
+    #${orig//[xyz]/_} ${~{accessibility_threshold}//./} ${~{mapq_threshold}//./}
+    #select_first(glob("~{output_dir}/median_coverage_by_GC_masked_*_~{sample_group_id}.csv"))
+    File logs = "calculate_mean_coverage_by_GC_~{sample_group_id}.log"
+    File coverage_output = "~{output_dir}/~{coverage_output_filename}"
+    File variance_output = "~{output_dir}/~{variance_output_filename}"
+    File output_gz = "~{output_dir}.tar.gz"
+  }
+}
+
+task CoverageHMM {
+  meta {
+    description: "Runs an mHMM on coverage counts obtained from a bamfile, using normalisation based on the mean coverage per GC bin over the whole genome of the individual. The outputs are normalized coverage values, 1 per window, per sample. Copy number state, 1 per window, per sample."
+  }
+  parameter_meta {
+    coverage_tarball: "The input tarball containing the coverage files"
+    gc_content_file: "The gc content file to use for the HMM calculations"
+    coverage_gc: "The coverage gc file to use for the HMM calculations"
+    coverage_variance: "The coverage variance file to use for the HMM calculations"
+    mapq_file: "The mapq file to use for the HMM calculations"
+    mapq_threshold: "The mapq threshold to use for the HMM calculations. Default is 0.5."
+    species: "The species of the samples specified in the manifest."
+    sample_id: "The sample name. This is typically the same as the input bam file name but without the extension."
+    docker: "(optional) the docker image containing the runtime environment for this task"
+    ram: "(optional) the amount of memory (MiB) to provision for this task"
+    cpu: "(optional) the number of cpus to provision for this task"
+    disk: "(optional) the amount of disk space (GiB) to provision for this task"
+    preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"
+    runtime_zones: "The ordered list of zone preference for running in GCP"
+  }
+  input {
+    File coverage_tarball
+    File gc_content_file
+    File coverage_gc
+    File coverage_variance
+    File mapq_file
+    Float mapq_threshold
+    String species
+    String output_dir
+    String sample_id
+    # runtime values
+    String docker = "us.gcr.io/broad-gotc-prod/cnv:1.0.0-1679431881"
+    String ram = "8000 MiB"
+    Int cpu = 16
+    # TODO: Make disk space dynamic based on input size
```

Comment:
same as above, do you still want to make disk space dynamic?

Reply:
These used a ridiculously small amount of disk (max 0.1 GiB), I think we are probably ok leaving this hard coded